provider.go

```diff
@@ -228,14 +229,22 @@ func (p *Provider) getSecret(token string, secretPath string, secretName string,
 		secretVersion = "0"
 	}
 
-	secretPrefix, secretSuffix := splitSecretPath(secretPath)
+	secretPrefix, secretSuffix := "", ""
```

Comment:
Instead of defaults, I think it would be better to fail if we don't have a prefix/suffix. Something like:

```go
s := regexp.MustCompile("/+").Split(secretPath, 3)
if len(s) < 3 {
	return "", fmt.Errorf("unable to parse secret path %q", secretPath)
}
secretPrefix := s[1]
secretSuffix := s[2]
```